lib/ip_util.c

```diff
@@ -131,9 +131,11 @@ int rc_get_srcaddr(struct sockaddr *lia, const struct sockaddr *ria)
 	}
 
 	if (connect(temp_sock, ria, SA_LEN(ria)) != 0) {
+		int e = errno; // Preserve connect's errno
 		rc_log(LOG_ERR, "rc_get_srcaddr: connect: %s",
```

Comment:
Should we do the errno preservation a documented property of this function? In that case we can have a unit testing of this function only if we cannot have an integration test.

Reply:
Good point. I could also change `rc_get_srcaddr()` to return `0` on success or `NETUNREACH_RC`/`ERROR_RC` on error. That may be a cleaner approach. What do you think?

Reply:
Makes sense to me